packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+		if (!super.persistValues()) return false;
+
+		this.getValueDB().setValue(
+			getDoorLockLoggingRecordsSupportedValueID(this.endpointIndex),
+			this.supportedRecordsNumber,
+		);
+
+		return true;
+	}
+}
+
+const convertEventTypeToLabel = (eventType: EventType): string => {
+	return eventTypeLabel[EventType[eventType] as keyof typeof EventType];
+};
+
+@CCCommand(DoorLockLoggingCommand.RecordsSupportedGet)
+@expectedCCResponse(DoorLockLoggingCCRecordsSupportedReport)
+export class DoorLockLoggingCCRecordsSupportedGet extends DoorLockLoggingCC {}
+
+@CCCommand(DoorLockLoggingCommand.RecordReport)
+export class DoorLockLoggingCCRecordReport extends DoorLockLoggingCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		validatePayload(this.payload.length >= 11);
+
+		this.recordNumber = this.payload[0];
+		this.recordStatus = this.payload[5] >>> 5;
+		if (this.recordStatus === RecordStatus.Empty) {
+			this.record = undefined;
+			this.recordTimestamp = undefined;
+			this.recordEventType = undefined;
+			this.recordUserID = undefined;
+			this.recordUserCode = undefined;
+		} else {
+			const dateSegments = {
+				year: this.payload.readUInt16BE(1),
+				month: this.payload[3],
+				day: this.payload[4],
+				hour: this.payload[5] & 0b11111,
+				minute: this.payload[6],
+				second: this.payload[7],
+			};
+
+			const eventType = this.payload[8];
+			this.recordUserID = this.payload[9];
+
+			// TODO: Parse User Code. The door lock I used does not provide
+			// 	     this data.
+			const userCode = undefined;
+
+			this.record = {
+				eventType: eventType,
+				label: convertEventTypeToLabel(eventType),
+				recordNumber: this.recordNumber,
+				timestamp: segmentsToDate(dateSegments).toISOString(),
+				userId: this.payload[9],
+				userCode,
+			};
+			this.recordTimestamp = segmentsToDate(dateSegments).toISOString();
+			this.recordEventType = eventType;
+			this.recordUserCode = undefined;
+		}
+
+		this.persistValues();
+	}
+
+	public readonly record: Record | undefined;
+	public readonly recordNumber: number;
+	public readonly recordStatus: RecordStatus;
+	public readonly recordTimestamp: string | undefined;
+	public readonly recordEventType: EventType | undefined;
+	public readonly recordUserID: number | undefined;
+	public readonly recordUserCode: string | Buffer | undefined;
+
+	public persistValues(): boolean {
+		if (!super.persistValues()) return false;
+
+		const valueDB = this.getValueDB();
+		const recordEventTypeID = getDoorLockLoggingRecordEventTypeID(
+			this.endpointIndex,
+		);
+		const recordNumberID = getDoorLockLoggingRecordNumberID(
+			this.endpointIndex,
+		);
+		const recordTimestampID = getDoorLockLoggingRecordTimestampID(
+			this.endpointIndex,
+		);
+		const recordUserIDID = getDoorLockLoggingRecordUserIDID(
+			this.endpointIndex,
+		);
+		const recordUserCodeID = getDoorLockLoggingRecordUserCodeID(
+			this.endpointIndex,
+		);
```

Comment:
This is going to cause problems when you request a record that is not the last one. The constructor (and by extension persistValues) is called for every report, including the ones that are returned in response to a GET. This would mean the values hold the last **received** record, but not necessarily the most recent one.

Reply:
I still think it is somewhat tricky to work with persisted values here as opposed to events. If I see it correctly, there's no way to know if a record is the last one, except when we know we are requesting the one with number 0. Or when handling an unsolicited report - but that's not yet known here.

In my head it makes sense to align this CC more with the Entry Control CC. When requesting a specific record via `commandClasses["Door Lock Logging"].get(n)` you get the correct one in response. And when the lock sends an unsolicited report, it would be emitted as an event the application can handle.

Reply:
Yeah, this was a very hacky way to do this, let me try another approach